base/array.jl

```diff
@@ -228,6 +228,8 @@ convert{T,n}(::Type{Array{T}}, x::Array{T,n}) = x
 convert{T,n}(::Type{Array{T,n}}, x::Array{T,n}) = x
 convert{T,n,S}(::Type{Array{T}}, x::Array{S,n}) = convert(Array{T,n}, x)
 convert{T,n,S}(::Type{Array{T,n}}, x::Array{S,n}) = copy!(similar(x,T), x)
+convert{T,n,S}(::Type{Array{T}}, X::AbstractArray{S, n}) = copy!(Array(T, size(X)), X)
```

Comment:
If you follow the definition of `convert{T,n,S}(::Type{Array{T}}, x::Array{S,n}` for consistency, you would define `convert{T,n,S}(::Type{Array{T}}, X::AbstractArray{S, n}) = convert(Array{T,n}, x)`. Written that way, I think the latter could even replace the former.